.github/workflows/reusable-setup-cache.yml

```diff
@@ -0,0 +1,73 @@
+# Description: this is a reusable workflow that sets up cache for unit and e2e testss.
+
+on:
+  workflow_call:
+    inputs:
+      commit:
+        # commit to run tests against
+        required: true
+        type: string
+      repository:
+        # repository to run tests against (possibly a fork of amplify-ui)
+        required: true
+        type: string
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Amplify UI
+        uses: actions/checkout@v3
```

Comment:
Didn't pin versions to specific SHAs in this PR. That'll be in a separate PR so that we can find and replace across the whole workflows 🙌